### PR TITLE
Dependency cleanup

### PR DESCRIPTION
**Pull request by wmontwe**
This is a first step to cleanup the project dependencies and declare transient dependencies better.

### CODE REVIEW

File: feature/widget/message-list/build.gradle.kts [hunk 1]
```diff
@@ -3,9 +3,13 @@ plugins {
 }
 
 dependencies {
+    api(projects.app.core)
```
Comment (cketti) [inline feature/widget/message-list/build.gradle.kts:6]: Why make `:app:core` part of `:feature:widget:message-list`'s API surface? Ideally a feature module would never make any of its dependencies part of its public API.

File: feature/widget/unread/build.gradle.kts [hunk 2]
```diff
@@ -3,13 +3,22 @@ plugins {
...
 dependencies {
-    implementation(projects.app.ui.legacy)
-    implementation(projects.app.core)
+    api(projects.app.core)
+    api(projects.app.ui.base)
+    api(projects.app.ui.legacy)
+    api(projects.mail.common)
 
-    implementation(libs.preferencex)
+    api(libs.preferencex)
```
Comment (cketti) [inline feature/widget/unread/build.gradle.kts:11]: Why make this particular library an API dependency?

**Comment by wmontwe** [comment]
This is a partial change to the dependencies setup and not complete. So there are `ìmplementation` to `api` changes that are only necessary for this intermediate step. Looking at the current priorities and time consumption to reach a final version, I will remove the `ìmplementation` to `api` changes and just keep the declaration of transient dependencies. 

The rest will be continued when there is more time available.

**Comment by wmontwe** [comment]
As this is not crucial at the moment, I'll close this and continue at a later point

**Pull request closed without merging**